src/lib.rs

```diff
@@ -1912,6 +1920,14 @@ impl Build {
                         cmd.push_cc_arg("-fno-plt".into());
                     }
                 }
+                if target.contains("-wasi") {
+                    // WASI does not support exceptions yet.
+                    // https://github.com/WebAssembly/exception-handling
```

Comment:
cc @thomcc I am ok with this `-fno-exception` as long as we only enable it for the wasi thst does not support it.

IIRC rust adds new preview targets for WASI with new features, so this could be fine ifcc checked the target?

Reply:
Yeah, probably fine for `-wasi` (e.g. preview 1), but eventually one of these will support exceptions and it'll be weird that we turn it off.

Reply:
Ok I've found it:

> Rust 1.78 will introduce new wasm32-wasip1 (tier 2) and wasm32-wasip2 (tier 3) targets. wasm32-wasip1 is an effective rename of the existing wasm32-wasi target, freeing the target name up for an eventual WASI 1.0 release.

https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html#:~:text=Rust%201.78%20will%20introduce%20new,an%20eventual%20WASI%201.0%20release.

Reply:
So the current wasm32-wasi will be renamed to wasm32-wasip1.

cc @antaalt can you change the target name please, to avoid breakage in the future?

cc @thomcc I think this is enough to prevent the breakage?